packages/salesforcedx-vscode-core/src/conflict/metadataCacheService.ts

```diff
@@ -49,6 +50,11 @@ export interface CorrelatedComponent {
   lastModifiedDate: string;
 }
 
+interface RecomposedComponent {
+  component?: SourceComponent;
+  children: Map<string, SourceComponent>;
```

Comment:
Why is component optional, but not children? Shouldn't this be the other way? We could simplify the code in line 283 just storing the sourcecomponent.

Reply:
The reason why component is optional is because we could have just the children and not the parent in the deploy. We could make the component non-optional but then I would want to add another field that is something like `parentIsInDeploy: boolean`.

Children could be optional. We would need to include a null check when adding to it, but that would be fine.